### PR TITLE
Handle Config.Deploy.elm from the remote repository

### PR DESCRIPTION
**Pull request by AronNovak**

### CODE REVIEW

File: .ddev/commands/web/gulp [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
 
+set -e
+
 ## Description: Run robo commands inside the web container for the client app.
 ## Usage: gulp
 ## Example: "ddev gulp publish" or "ddev gulp".
 
+CLIENT_PATH="../../client"
+
+if [[ -z "${EHEZA_INFRA_REPO_REMOTE}" ]]; then
```
Comment (anvmn) [inline .ddev/commands/web/gulp]: @AronNovak shouldn't we do all this only when running `gulp publish`?

Comment (AronNovak) [reply .ddev/commands/web/gulp]: good point, added a condition for that

File: .ddev/commands/web/gulp [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+  echo "Cloning eheza-infra repo..."
+    git clone "${EHEZA_INFRA_REPO_REMOTE}" ../../eheza-infra
+  else
+    echo "Updating eheza-infra repo..."
+    cd ../../eheza-infra
+    git pull
+    cd -
+  fi
+
+  cp ../../eheza-infra/elm/Config.Deploy.elm ${CLIENT_PATH}/src/elm/
```
Comment (anvmn) [inline .ddev/commands/web/gulp:28]: This means that we need to have that `Config.Deploy.elm` under `elm` folder.
Maybe it should be under `config`, or maybe even under root?

File: README.md [hunk 3]
```diff
@@ -78,14 +78,15 @@ The default method assumes Pantheon as the hosting service provider.
 Prepare `Config.Deploy.elm` based on `Config.elm` that holds the
 infrastructure-related data for Elm. This file is gitignored, and it is
 used during Elm compilation, before the final artifact is pushed to Pantheon.
-The source of truth for this deployment config is in the Pantheon artifact repository,
-can be found under the repository root, so most of the time, you can copy it from there.
+Put it in a private repository, and link that repository using
```
Comment (anvmn) [inline README.md:81]: Need to mention where `Config.Deploy.elm`  should be located.

**Review by anvmn — COMMENTED** [review]
@AronNovak 2 comments

File: .ddev/commands/web/gulp (hunk 2 above)
Comment (AronNovak) [reply .ddev/commands/web/gulp:28]: https://github.com/Gizra/ihangane-infra/pull/11/files - as it's multi-purpose repository, I'd not put it under root.
With `elm`, we distinguish it from backend-ish things.

File: README.md (hunk 3 above)
Comment (AronNovak) [reply README.md:81]: Good point, adding a sentence on that.

Comment (AronNovak) [reply README.md:81]: done

**Review by anvmn — APPROVED** [review]

**Pull request merged**